q2_composition/_dataloaf_tabulate/_visualizer.py

```diff
@@ -45,11 +45,21 @@ def tabulate(output_dir: str, data: DataLoafPackageDirFmt):
         slice_contents.append(slice_html)
 
     slice_tables = zip(slice_names, slice_contents)
+
     # Filling in the table that will appear on index.html
-    context = {
-        'intercept': slice_md_json['metadata']['intercept_groups'],
-        'tables': slice_tables
-    }
+    intercept = slice_md_json['metadata']['intercept_groups']
+    if isinstance(intercept, str):
```

Comment:
Not sure if there's a better way to handle this - when the intercept only contains one term, it's a `str`, and when there are multiple terms it's a `list`. This approach prevents the weird string splitting for the intercept contents when there's only one term.

Reply:
We actually do this sort of thing pretty regularly. [Here](https://github.com/qiime2/qiime2/blob/fe49136357225e47241b2727883d75390059b7bc/qiime2/sdk/action.py#L359) and on line 440 of the same file we call a util function called [tuplize](https://github.com/qiime2/qiime2/blob/fe49136357225e47241b2727883d75390059b7bc/qiime2/core/util.py#L40) that just makes it so if an action only returned a single value we wrap it in a tuple so we know we will always have an iterable even if it's an iterable of one item.

Reply:
I've also personally accidentally iterated over a string in this kind of way about a million times.

Reply:
Oh okay cool, that's reassuring! I'll leave as-is then since this should end up being a list object (even if it's a list of just one item).